doc/cabal-package.rst

```diff
@@ -1574,7 +1574,7 @@ in ghc's package DB and so we can figure out what the location of the library
 is. Foreign libraries however don't get registered, which means that we'd have
 to have a way of finding out where a platform library got installed (other than by
 searching the ``lib/`` directory). Instead, we install foreign libraries in
-``~/.cabal/lib``, much like we install executables in ``~/.cabal/bin``.
+``~/.cabal/lib``.
```

Comment:
No, `~/.config/cabal/lib` would be wrong.  In an ideal world, it should be somewhere in `$XDG_STATE_HOME` I think.  But the impression I got is that this specific behaviour for installing foreign libraries is legacy, and only used for exotic instances of v1-builds.  I did not change the code for this, so they will indeed install into `~/.cabal/lib`.  If I am wrong, then this is something that should definitely be addressed.

Reply:
I confused this with something else (the InstallDirs business).  Now I'm actually curious where these foreign libraries get installed, since they haven't shown up in any of my tests.  I wonder if they end up in `~/.local/lib` now, since I don't remember seeing any code dealing with them.  Trying to find a buildable example that uses them...

Reply:
Oh, I guess it *is* currently v1-install only:  #6046

But with this patch, v1-install will put the library in `~/.local/lib`, which seems reasonable.

Reply:
Just FYI: we still have plenty of v1 users (I'd add: unfortunately). Please, make sure this will not break for them (at least, not easily :-)).

Reply:
It doesn't break; I just tested it.  If `~/.cabal` exists then the library is put in `~/.cabal/lib`.

Reply:
Have you tried to use a v1-installed library that war put in .local/lib?

Also, if.local/lib may be used, is it documented anywhere? I don't see it.

Reply:
"Use" in what sense?  Aren't foreign libraries for use by C programs that are not themselves built by Cabal?

Reply:
> I confused this with something else (the InstallDirs business). Now I'm actually curious where these foreign libraries get installed, since they haven't shown up in any of my tests. I wonder if they end up in `~/.local/lib` now, since I don't remember seeing any code dealing with them. Trying to find a buildable example that uses them...

In case you are still searching, here are some examples: cabal-testsuite/PackageTests/ForeignLibs

Reply:
I managed to cobble a test together from the documentation itself, which is how I confirmed that it now installs to `~/.local/lib` unless `~/.cabal` exists.

Reply:
Okay, I read up a bit more on this, and I think it is fine.